test/wordsearch.js

```diff
@@ -0,0 +1,79 @@
+/* global describe, it */
+
+const assert = require('assert');
+const { WordSearchSolver } = require('../');
+
+describe('WordSearchSolver', function () {

```

Comment:
I don't see a test for the case where there are two words in the list which overlap in the same direction, e.g. "foot" and "football".  It looks like the code handles it, but we should probably validate it.